example/src/App.tsx

```diff
@@ -373,8 +378,27 @@ const App = () => {
           isNativeUIMRecShowing &&
             <AppLovinMAX.AdView adUnitId={MREC_AD_UNIT_ID}
                                 adFormat={AppLovinMAX.AdFormat.MREC}
-                                style={styles.mrec}/>
-        }
+                                style={styles.mrec}
+                                onAdLoaded={(adInfo) => {
+                                  setStatusText('MREC ad loaded from ' + adInfo.networkName);
+                                }}
+                                onAdLoadFailed={(errorInfo) => {
+                                  setStatusText('MREC ad failed to load with error code ' + errorInfo.code + ' and message: ' + errorInfo.message);
+                                }}
+                                onAdClicked={(adInfo) => {
+                                  setStatusText('MREC ad clicked');
+                                }}
+                                onAdExpanded={(adInfo) => {
+                                  setStatusText('MREC ad expanded')
+                                }}
+                                onAdCollapsed={(adInfo) => {
+                                  setStatusText('MREC ad collapsed')
+                                }}
+                                onAdRevenuePaid={(adInfo) => {
```

Comment:
Can you test omitting this callback (not sure if possible) and seeing if iOS will crash?

Reply:
Hmm ok, [this LoC](https://github.com/AppLovin/AppLovin-MAX-React-Native/pull/151/files#diff-bde49a87cdd47f60e5f842aad0e448a5b2957a97f551a40b9ceb0474efef48fbR349) will crash if `self.onAdRevenuePaidEvent` is `nil`.

Reply:
Now I got it.   Since [the JavaScript function](https://github.com/AppLovin/AppLovin-MAX-React-Native/pull/151/files#diff-c5a5de072365504a51cc94ecc502ec06c40cec02acc6eafdab7ef56433933897R122) is always defined and passed to the native side, it won't crash.   But, if it is removed, it will crash since it becomes `nil`.    We can check if `self.onAdRevenuePaidEvent` or all the other JS functions are not `nil`.

Reply:
I don't think it's necessary to do the `nil` check then, if we always define/pass it. I was only worried that if the publisher did not implement the function, we won't pass anything and `self.onAdRevenuePaidEvent` will be `nil`.